### PR TITLE
Get blog by category

### PR DESCRIPTION
**Pull request by cp-priyanka-g**
- [x] Get post by category.
- [x] Add tags in post.
- [x] Get Blog By Tag.
- [x] Handle the error status

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export default Object.freeze({
-  STRAPI_URL: "https://blog-admin.canopas.com",
+  STRAPI_URL: "http://localhost:1337",
```
Comment (cp-shruti-s) [inline config.js:2]: Remove the local URL

File: lib/post.js [hunk 2]
```diff
@@ -34,4 +42,4 @@ async function formateDate(date) {
   return formattedDate;
 }
 
-export { fetchPost, getReadingTime, formateDate };
+export { fetchPost, getReadingTime, formateDate,fetchCategory};
```
Comment (cp-shruti-s) [inline lib/post.js:45]: Merge with master

File: pages/categories/[slug].js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+export async function getStaticProps(context) {
+    const slug = context.params.slug;
+
+  if (!slug) {
+    throw new Error("Slug not valid");
+  }
+  
+  var category = await fetchCategory(slug);
+  console.log(category)
```
Comment (cp-shruti-s) [inline pages/categories/[slug].js:96]: remove console log

File: pages/categories/[slug].js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  for (let i = 0; i < category.data.length; i++) {
```
Comment (cp-shruti-s) [inline pages/categories/[slug].js:97]: If is it not necessary then remove

File: pages/categories/[slug].js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                    <MarkdownView
+                      className="text-gray-800 mt-5 text-sm line-clamp-3"
+                      markdown={post.content}
+                    />
+                    <div className="pt-4">
+                      <div className="flex flex-row ">
+                        <div className="text-gray-500 flex">
+                          <span>{post.publishedAt}</span>
+                          <span className=" after:content-['\00B7'] after:mx-1 "></span>
+                          <span>{post.readingTime}</span>
```
Comment (cp-shruti-s) [inline pages/categories/[slug].js:61]: How to calculate post reading time ??

File: pages/index.js [hunk 6]
```diff
@@ -98,6 +105,6 @@ export async function getStaticProps() {
   }
 
   return {
-    props: { posts },
+    props: { posts},
```
Comment (cp-shruti-s) [inline pages/index.js:108]: please configure prettier for the code format

File: lib/post.js [hunk 7]
```diff
@@ -44,4 +34,4 @@ async function formateDate(date) {
   return formattedDate;
 }
 
-export { fetchPost, getReadingTime, formateDate, fetchAuthor };
+export { fetchPost, getReadingTime, formateDate, fetchAuthor,fetchCategory};
```
Comment (cp-shruti-s) [inline lib/post.js:37]: Where is the category api configuration method ??

File: lib/post.js [hunk 8]
```diff
@@ -6,8 +6,18 @@ const WORDS_PER_MINUTE = 200;
 // GET the posts and post by id from Strapi
 async function fetchPost(slug) {
   const requestUrl = slug
-    ? `${STRAPI_URL}/v1/posts/${slug}?populate=deep`
-    : `${STRAPI_URL}/v1/posts?populate=deep`;
+    ? `${STRAPI_URL}/api/posts/${slug}?populate=*`
```
Comment (cp-sumi-k) [inline lib/post.js:9]: Make sure that other code should not be changed. `*` was `deep`

File: lib/post.js [hunk 9]
```diff
@@ -6,8 +6,18 @@ const WORDS_PER_MINUTE = 200;
...
+    : `${STRAPI_URL}/api/posts?populate=*`;
+  const response = await fetch(requestUrl);
+  const data = response.json();
+  return await data;
+}
+
+// GET the posts and post by id from Strapi
+async function fetchCategory(slug) {
+  const requestUrl = slug
+    ? `${STRAPI_URL}/api/categories/${slug}?populate=*`
```
Comment (cp-sumi-k) [inline lib/post.js:19]: `*` can be `deep` to get full response

File: pages/categories/[slug].js [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+            );
+          })}
+        </div>
+      </div>
+    </section>
+  );
+};
+
+export async function getStaticPaths() {
+  const Posts = await fetchCategory();
```
Comment (cp-sumi-k) [inline pages/categories/[slug].js:72]: Give relative name to const

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

**Comment by cp-shruti-s** [comment]
Screenshot from 2022-11-29 14-38-03

Refer the more blog website because tags ui is not suitable

File: pages/categories/[slug].js [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+    <section className="py-5">
+      <div
+        className="container flex flex-col sm:px-[4rem] md:px-[8rem] lg:px-[10rem] xl:px-[12rem] 2xl:px-[17rem]"
+        key={category.id}
+      >
+        <a
+          href="/"
+          className="text-pink-400 flex justify-center text-4xl sm:text-7xl mt-10 mb-20"
+        >
+          Blog By Category
```
Comment (cp-shruti-s) [inline pages/categories/[slug].js:25]: Header should be common in website

File: pages/index.js [hunk 12]
```diff
@@ -190,11 +197,14 @@ export async function getStaticProps() {
   if (posts.data) {
     for (let i = 0; i < posts.data.length; i++) {
       const post = posts.data[i].attributes;
+
+      for (let i = 0; i < post.tags.data.length; i++) {
+        var tag = post.tags.data[i].attributes;
```
Comment (cp-shruti-s) [inline pages/index.js:202]: this variable is used anywhere ??

**Review by cp-shruti-s — APPROVED** [review]

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**